src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -43,7 +43,7 @@ public String tagColor() {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
             || (other instanceof Tag // instanceof handles nulls
-            && tagName.equals(((Tag) other).tagName)); // state check
+            && tagName.equalsIgnoreCase(((Tag) other).tagName)); // state check
```

Comment:
Puts all the equals method from different tag types in one location and inherit from its parent class, a better coding quality.